### PR TITLE
fix: Handle parsing errors

### PR DESCRIPTION
**Pull request by andreas-unleash**
Catch any errors from parsing toggles from storage
Relates to #39

### CODE REVIEW

File: lib/parse_stringify_toggles.dart [hunk 1]
```diff
@@ -5,12 +5,23 @@ import 'package:unleash_proxy_client_flutter/toggle_config.dart';
...
-  final data = jsonDecode(body)['toggles'];
-  // Check if there is anything to map over? Otherwise map might cause an error
-  // Write a test that checks if the
-  return {
-    for (var toggle in data) toggle['name']: ToggleConfig.fromJson(toggle)
-  };
+  var decoded = jsonDecode(body);
+
+  // Return an empty map if 'toggles' isn't a list or doesn't exist
+  if (decoded['toggles'] is! List) {
```
Comment (sighphyre) [inline lib/parse_stringify_toggles.dart:11]: I'd really rather avoid checking for a list here. Stuff should fail and the new catch block should catch that, right? I don't really see a reason this function needs to change, it looks like it's correct, it's getting fed garbage data

Comment (andreas-unleash) [reply lib/parse_stringify_toggles.dart:11]: sure makes sense - this was my initial fix, but the try/catch should be enough here agreed

Comment (sighphyre) [reply lib/parse_stringify_toggles.dart]: Nice! I think this looks good, but maybe we keep this open until we understand more about this problem. I'd need to go root around a little in the code

**Review by sighphyre — APPROVED** [review]
Yeah let's do the thing

**Pull request merged**